lldb/include/lldb/Target/SwiftLanguageRuntime.h

```diff
@@ -256,6 +256,10 @@ class SwiftLanguageRuntime : public LanguageRuntime {
   llvm::Optional<unsigned> GetNumChildren(CompilerType type,
                                           ValueObject *valobj);
 
+  llvm::Optional<size_t> GetIndexOfChildMemberWithName(
+      CompilerType type, const char *name, ExecutionContext *exe_ctx,
+      bool omit_empty_base_classes, std::vector<uint32_t> &child_indexes);
+
```

Comment:
This looks good — but it would be better to use an `llvm::StringRef` instead of a `const char*` for `name`.

Reply:
Similarly, a MutableArrayRef http://llvm.org/doxygen/classllvm_1_1MutableArrayRef.html is preferred over a std::vector&. Then call sites can also use SmallVectors if they wish to do so.

Reply:
thanks will do! TIL about `MutableArrayRef`

Reply:
done

Reply:
Seems `MutableArrayRef` won't work here, I don't see any way to modify an underlying vector, in particular to append to it.

Reply:
`MutableArrayRef` seems like a mutable view API over some block of non-changing underlying storage.

Reply:
Whoops, you're right! We could go to a `SmallVectorImpl<uint32_t>` but the returns are marginal.

Side note: Having now read the documentation for `child_indexes` (https://lldb.llvm.org/cpp_reference/classlldb__private_1_1CompilerType.html#abbeabd29258512dc7869c3f01afff93d) this method's entire API is _weird_.

Reply:
haha very weird

Reply:
For posteriority: The point of the child_indexes array is to support returning more than one match. I have no idea in which language this would even be possible. To top that off, the return value is that of child_indexes[0] or INT_MAX, and the comment has a TODO mark in it. We might want to check if this is even wired up in _any_ language backend and otherwise potentially even remove it.

Reply:
sounds good to me.

In the case of multiple children with the same name, a class can have a property of the same name as a property in its base class if the base class property is private.

In my quick check of c++, it allows duplicate members names regardless of access control.

Reply:
It really depends on what we decide to mean by "ChildMember".  If it means any member of the current object, then in C++ you can have two with the same name since they can come from different base classes.  If it only means "any member contributed by the most specific class of the object, and you have to get to the others by recursing through the base class by hand, then I can't think of a way you could have two members with the same name.

I think the first meaning is more useful, but I'm not 100% set on that.